### PR TITLE
Async

### PR DESCRIPTION
**Pull request by johanot**

### CODE REVIEW

**Comment by srhb** [comment]
I pushed some commits. I think it _eventually_ works most of the time now :tm: 

- I think the touch-rewrite was only half-done, think i fixed it in a sensible way.
- I fixed metrics so it doesn't spawn a runtime (inside the runtime)
- I enabled the secondary vault monitor config test.

There's some issues I am still debugging before I think it's ready for review by someone wiser than me. Writing them down here so I can keep track of where I am.

1. something is spamming
```
vm-test-run-faythe-vault-test> client # [   32.872869] faythe[881]: [2024-10-18T08:52:39Z ERROR rustify::client] error=Server returned error
vm-test-run-faythe-vault-test> client # [   32.873456] faythe[881]: [2024-10-18T08:52:39Z ERROR rustify::endpoint] error=Server returned error
vm-test-run-faythe-vault-test> client # [   32.874041] faythe[881]: [2024-10-18T08:52:39Z ERROR vaultrs::api] Detected errors in API response: []
vm-test-run-faythe-vault-test> client # [   32.874639] faythe[881]: [2024-10-18T08:52:39Z ERROR vaultrs::kv2] error=The Vault server returned an error (status code 404)
```

Haven't found out what yet. Probably part of the raw cert check.

2. faythe appears to set up too many challenges according to pebble. I've seen 5 challenges in the db at one time.

3. faythe appears to set up the wrong dns proofs sometime, getting stuck validating internally. Unsure why this happens.

4. sometimes there's nonce errors in the AcmeClient. I think this may simply be because of the default pebble setup `vm-test-run-faythe-vault-test> acme # [    5.144768] pebble[662]: Pebble 2024/10/18 08:52:12 Configured to reject 5% of good nonces` -- but we should handle this (and maybe that'll fix some other issues? I do think I'm seeing them too often for the 5% to be the only reason)

File: src/vault.rs [hunk 1]
```diff
@@ -371,26 +371,23 @@ impl VaultSpec {
...
-
-        let raw_read: Result<VaultData, ClientError> =
-            kv2::read(&*client, &persist_spec.kv_mount, &persist_spec.paths.meta).await;
-
-        match raw_read {
-            Ok(value) => {
-                let time_stamp = chrono::DateTime::parse_from_rfc3339(&value.borrow())?;
-                let diff = chrono::Utc::now().signed_duration_since(time_stamp);
-                match diff
-                    > chrono::Duration::milliseconds(
```
Comment (johanot) [inline src/vault.rs]: @srhb ... understand you fixed some issue with "raw_read", but... it looks like - from this diff - that the logic handling should_retry time-diff totally disappeared? Causing "RecentlyTouched" to never be constructed. I would imagine this causing a lot of concurrent issuing attempt. Maybe this explain some of the other problems you are describing?

Comment (johanot) [reply src/vault.rs]: Having #95 could have CI-caught the unused error variant :)

**Comment by srhb** [comment]
I've fixed the remaining _known_ bugs in our fork of acme_lib (tmp commit added while the fork change gets approved) -- at least I can now consistently run the test without failures...

File: src/vault.rs (hunk 1 above)
Comment (srhb) [reply src/vault.rs]: :zombie_woman: wdyt, did I fix it, or should I go back and try to recreate from where you were? It might be hard to follow at this point, months later :sweat:

File: src/monitor.rs [hunk 2]
```diff
@@ -164,120 +169,120 @@ mod tests {
...
         assert_eq!(spec.name, name);
         assert_eq!(
             spec.cn.to_domain_string(),
             String::from("*.subdivision.unit.test")
         );
-        thread.join().unwrap();
     }
 
-    #[test]
+    //#[test]
```
Comment (cafkafk) [inline src/monitor.rs:205]: If we comment out this test, we should at least give an explanation for why where doing it, and when we can enable it again

**Review by cafkafk — CHANGES_REQUESTED** [review]

File: src/monitor.rs (hunk 2 above)
Comment (srhb) [reply src/monitor.rs:205]: OK, I'll try to figured out why it bork.

Comment (cafkafk) [reply src/monitor.rs:206]: I think removal may make more sense given it's never worked anyways, but that can be a follow up commit.

**Review by cafkafk — APPROVED** [review]

File: src/vault.rs (hunk 1 above)
Comment (cafkafk) [reply src/vault.rs]: Considering this PR is blocking for any other changes, I think it's reasonable to resolve this conversation and then corrections can be made in a follow up PR. Specially considering johanots point that certain checks may have caught this in the first place, those checks can only come after we merge this.

May also be a cautionary tale of not sending large patchsets in the future, smaller merges faster.
